templates/nextjs/src/hooks/useActiveWallet.tsx

```diff
@@ -0,0 +1,42 @@
+import { useEffect, useState } from "react";
+import { useBrowserWallet } from "./useBrowserWallet";
+import { useBurnerWallet } from "./useBurnerWallet";
+
+/**
+ * burner -> a burner wallet embedded inside of the template app and stored in local storage
+ * browser -> a wallet connected via a browser extension like the Fuel Wallet
+ */
+type WalletTypes = "burner" | "browser";
+
```

Comment:
Is it worth these hooks sharing a common interface?

**Example**
```ts
interface Wallet {
  wallet: WalletLocked;
  walletBalance: number;
  refreshWalletBalance: number;
}

interface BrowserWallet extends Wallet {
  isWalletConnected: boolean;
  walletNetwork: string; // Change me to correct type
}
```

**Usage**
```ts
export const useActiveWallet = () => {
  const burner: Wallet = useBurnerWallet();
  const browser: BrowserWallet = useBrowserWallet();

  return activeWallet === "browser" ? browser : burner;
}
```

Reply:
Great suggestion. https://github.com/FuelLabs/fuels-ts/pull/1873/commits/941b09e1f00664ba57a6651751c5b30e21c7a9bd